views/js/ps_event_listener.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ */
+
+const allPrestaShopJSEvents = [
+  'handleError',
+  'showErrorNextToAddtoCartButton',
+  'updateProductList',
+  'updateProduct',
+  'responsive',
+  'updateProductQuantityInCart',
+  'updateFacets',
+  'updateCart',
+  'updatedCart',
+  'clickQuickView',
+  'changedCheckoutStep',
+  'editAddress',
+  'updatedAddressForm',
+  'updatedDeliveryForm',
+  'editDelivery',
+];
+
+$(document).ready(function () {
+
+  allPrestaShopJSEvents.forEach(function(eventName) {
+    const listenerParameters = '&fc=module&module=ps_qualityassurance&controller=listener&action=RecordJSHookExecution&ajax=true';
+
+    prestashop.on(eventName, function(event) {
+      $.post({
+        url: '/?name='+eventName+listenerParameters,
```

Comment:
you can use shop url from `prestashop.url` because this will fail if some reason installation is in subdirectory or for example virtual uri multi-store